src/hidpp20.h

```diff
@@ -317,6 +324,46 @@ union hidpp20_button_binding {
 } __attribute__((packed));
 _Static_assert(sizeof(union hidpp20_button_binding) == 4, "Invalid size");
 
+struct hidpp20_color {
+	uint8_t red;
+	uint8_t green;
+	uint8_t blue;
+};
+
+// TODO: put the leds into array and access them through this enum
+enum hidpp20_led_type {
+	HIDPP20_LED_UNKNOWN = -1,
+	HIDPP20_LED_LOGO = 0,
+	HIDPP20_LED_SIDE
+};
+
+struct hidpp20_internal_led {
+	uint8_t mode;                  /* 00 - off; 01 - on; 03 - cycle; 0a - breath */
+	struct hidpp20_color color;
+	union {
+		struct {
+			uint8_t free[2];
```

Comment:
uint16_t please. Avoid pad arrays unless where there are no good datatypes. Even for the free[3] below please use a uint8_t and a uint16_t. Makes the size more obvious.

you could be using _Static_assert for these too if you want